### ISSUE TITLE
Question about refining the model

### ISSUE THREAD

**Issue by zroger49** [bowang-lab/IntegrAO#15]
Hello! Thank you for your work, IntergrAO has consistently been scoring very well on our internal benchmarks! 
I have a question regarding the fine-tunning the model. 
In the paper (at least in the pre-print), you mention that 

"After unsupervised integration of multi-omics data, patient subtypes can be determined by clustering the final integrated network. Given defined subtype labels, IntegrAO can be further fine-tuned to predict subtypes for new patients based on any combination of omics data" 

Since clustering is an unsupervised method, with ground labels usually unknown, my question is that is supposed to be used as "defined subtype labels". Lets take a cancer as an example. Usually there are different sets of subtypes, depending on which data type was analyzed, and in many cases I believe there is no gold-standard (I could be wrong). 

1) use the labels as output of the model -> I only see this making sense, if I see that the clusters are clearly separating the data, based on different omics. However, this requires extensive validation, but could be used to predict my new subtypes based on just a subset of omics.
2) Use already defined subtypes, even if they are based on only one data modality?

THank you for your time. 

PS: I do not have access to the final paper, is there any chance you could email to me?

**Comment by rexxxx1234** [comment]
Hi @zroger49,

Thank you for your interest in IntegrAO and glad to hear it's working well!

You are absolutely right that there is often no gold standard set of subtypes when integrating multi-omics data for cancer. Many of the existing subtype definitions are validated but are often biased toward one specific modality from which they were derived, such as transcriptomics.

The main idea behind IntegrAO is that integration across multiple omics modalities should lead to more representative and robust patient clusters. For that reason, we use the clusters derived from the integrated network as labels for fine-tuning, as they aim to capture a more holistic biological signal.

That said, you are welcome to fine-tune IntegrAO using any set of subtype labels you have, even if those labels are based on a single modality. The model will still learn to predict those subtypes from any subset of omics data. However, doing so may not fully utilize the integrative strength of the framework, since the supervision is aligned with only one view of the data.

Ultimately, the decision depends on your specific use case and how you define the problem you are working on.

Feel free to email me regaridng the copy of the final paper: rex.ma@mail.utoronto.ca

Best,
Rex

**Comment by zroger49** [comment]
Thank you for your comment. As a follow-up, since many problems do not have defined subtypes for multi-modal data, would it make sense to use the output labels from a IntergrAO run to refine the model? Or is this analysis circular in nature? 

Best, 
Rogério

**Comment by rexxxx1234** [comment]
Could you clarify what do you mean by "refine" the model? The fine-tuning steps for IntegrAO is to extend the model for subtype prediction, not further integration. The intended use case is to predict subtypes for new samples based on any given combination of omics data.

Regarding the absence of predefined subtypes: in the manuscript, we demonstrated that IntegrAO can integrate partial multi-omics data and uncover biologically and clinically relevant subtypes with all the downstream validation. If your goal is to define potential new subtypes using multi-modal data, such validation will be necessary.

If you want to provide the research question you’re working on, I can provide more helpful suggestions.

**Comment by zroger49** [comment]
Apologies for the late reply. 
What I meant with "refine" is if I could use the subtype labels, outputted by IntergrAO (Running K-means or spectral clustering on the final embedding), to extend the model for classification. 

My research question is more technical, as I am benchmarking different methods and I would like to highlight advantages and features across different packages. 

My proposed pipeline is the following: 

1. Run IntergrAO on the train samples. Uncover subtypes using spectral on the final embedding. 
2. Use these labels to fine tune the model for prediction, i.e. 
```
embeds_final, S_final, model, preds = integrater.classification_finetuning(spectral_labels, result_dir, finetune_epochs=800)
```
3. Using the test samples, which are partial multi-omics samples, obtain the subtype predictions for this samples?

Does this make sense?

**Comment by rexxxx1234** [comment]
This approach definitely makes sense if your goal is to leverage the multi-omic data to generate subtype predictions on the test samples.

That said, since you mentioned this is for benchmarking, I’m curious whether your dataset provides ground-truth subtype labels. If so, using those for classification fine-tuning might give a more consistent and comparable baseline across different methods.

**Comment by rexxxx1234** [comment]
mark it as close for now, feel free to reopen if needed!

**Comment by zroger49** [comment]
Thank you for your answer. 
Regarding the benchmark, I'm testing subtyping and not classification into new groups, hence I think using gold-truth labels (which are often based on one data modality) would not make that much sense. 

You can keep this issue closed.

**Comment by rexxxx1234** [comment]
I agree
